ansible/roles/nginx/defaults/main.yml

```diff
@@ -492,7 +496,10 @@ nginx_ssl_dhparam_set: 'default'
 #
 # Default set of cipher suites to use.
 # Refer to ``nginx_ssl_ciphers`` for details.
-nginx_default_ssl_ciphers: 'bettercrypto_org__set_b_pfs'
+nginx_default_ssl_ciphers: '{{ "mozilla_modern"
+                               if nginx_default_tls_protocols|length == 1 and
+                                  nginx_default_tls_protocols[0] == "TLSv1.3"
```

Comment:
This would cause a fallback to the cipher suites in TLS 1.3, which includes `TLS_AES_128_CCM_8_SHA256`. [NCSC-NL](https://english.ncsc.nl/publications/publications/2019/juni/01/it-security-guidelines-for-transport-layer-security-tls) explicitly considers this cipher to be insufficient because it is "a variant of AES-CCM with a truncated authentication tag, which has a lower security equivalent for integrity protection."

Reply:
`openssl ciphers 'HIGH:!aNULL:!MD5'` gives me a long list of ciphers, including these CCM8 ciphers:
```
ECDHE-ECDSA-AES256-CCM8
DHE-RSA-AES256-CCM8
ECDHE-ECDSA-AES128-CCM8
DHE-RSA-AES128-CCM8
DHE-PSK-AES256-CCM8
AES256-CCM8
PSK-AES256-CCM8
DHE-PSK-AES128-CCM8
AES128-CCM8
PSK-AES128-CCM8
```
This is with OpenSSL version 1.1.1d-0+deb10u3.

Reply:
> openssl ciphers 'HIGH:!aNULL:!MD5' gives me a long list of ciphers, including these CCM8 ciphers:

I see, following Mozilla recommendation to the letter is not suitable for debops, where options are more finegrained. When you enable `mozilla_modern` cipher suite, debops does not forbid to use TLSv1.2, even though Mozilla modern recommendation is TLSv1.3-only (which would defang the CCM8 ciphers).

The security hole needs to be fixed along with the usability problem. I'm currently trying to create a cipher suite name that `openssl ciphers` expands to just TLSv1.3 ciphers, which can then be used for the `mozilla_modern` value in `nginx_ssl_ciphers`. Nginx should then fail if configured with TLSv1.2-only (test pending).

Reply:
I haven't found (quickly) how to specify disabling all TLSv1.2 and below ciphers. The webserver error could also be entirely avoided, if I just copy `nginx_ssl_ciphers.mozilla_intermediate` to `mozilla_modern`. This would not change a thing for TLSv1.3-only, while TLSv1.2-only or dual would not have the usability bug and potential security hole. If you haven't any more concerns, I'll go with copying.